### PR TITLE
Create docusaurus files

### PR DESCRIPTION
**Pull request by ivanha09**
Created a new pull request with your @ybiquitous suggestions of putting them into the master branch and putting it under the docusaurus folder. 
I included the yarn.lock and ignored the package-lock.json.

### CODE REVIEW

**Comment by ybiquitous** [comment]
Related to #65

File: docusaurus/docs/rules.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+id: rules
+title: Rules
+sidebar_label: Rules
+---
+
+
+## Rule: "_blank" Security Issue
+When `target = "_blank"` is used, the opened page can access the original window object and potentially redirect the original page to a malicious URL. In this example, the rule will look for patterns of `"_blank"` and suggest to use `rel="noopener"` to prevent the opened page from having access.
```
Comment (ybiquitous) [inline docusaurus/docs/rules.md]: Umm..., this syntax highlighting seems to be broken. If adding a blank line as follows, does it work?

```
## Rule: "_blank" Security Issue

When `target = "_blank"` is used, ...
```

Comment (ivanha09) [reply docusaurus/docs/rules.md]: > Are `versioned_*` directories auto-generated?

Yes. They are generated when the script `yarn run version 1.0.0` is run.
https://docusaurus.io/docs/en/versioning

**Comment by ivanha09** [comment]
> Are `versioned_*` directories auto-generated?

> Are `versioned_*` directories auto-generated?

Yes. They are generated when the script `yarn run version 1.0.0` is run.
https://docusaurus.io/docs/en/versioning

**Comment by ybiquitous** [comment]
I see. It seems better that we don't commit the auto-generated versioned directories... 🤔 

Maybe we may be able to use an auto-deployment script:
https://docusaurus.io/docs/en/publishing#automating-deployments-using-continuous-integration

For example:

```shell
cd website && yarn install && yarn run version $GIT_TAG && yarn run publish-gh-pages
```

**Comment by ivanha09** [comment]
I tried to test the site after deleting the auto-generated versioned
directories but it doesn't seem to work if I don't have them in there. I
don't think they will repopulate I don't include them so I think they have
to be kept in there.

When talking with Matsumoto we just went with the GitHub pages, but I can
try setting up an auto-deployment script. Should I try it with Circle CI?

**Comment by ybiquitous** [comment]
Sorry, let's see the Docusaurus repository itself! It seems to commit `versioned_*` directories. 😅 
https://github.com/facebook/docusaurus/tree/master/website-1.x

image

**Comment by ybiquitous** [comment]
If starting auto-deploy, it seems good to me to use Travis CI which we use now. It's easy.

**Comment by ivanha09** [comment]
> Sorry, let's see the Docusaurus repository itself! It seems to commit `versioned_*` directories. 😅
> https://github.com/facebook/docusaurus/tree/master/website-1.x
> 
> image

Thanks for checking. That's a good way to check. 👍

**Comment by ivanha09** [comment]
@ybiquitous I was looking at Travis CI. It looks like it's already installed on this repo so I think I would only need to include [step 5](https://docusaurus.io/docs/en/publishing#using-travis-ci ) into the .travis.yml file.

If that is all. Should I include it in this pull request?

**Comment by ybiquitous** [comment]
Nice. 👍
But I recommend to try it on another pull request. I think it better to restrict the scope of a pull request. Many diffs mak humans more tired. 😅

**Pull request merged**